### PR TITLE
Add comments to MemAccessUtils.h.

### PR DESCRIPTION
**Pull request by atrick**
The name doesn't mean "memory access" colloquially. It specifically
refers to memory operations that take part in formal access scopes
marked by begin_access/end_access instructions. If you aren't working
with those instructions it really doesn't make sense to use it.

### CODE REVIEW

**Comment by atrick** [comment]
@swift-ci smoke test

File: include/swift/SIL/MemAccessUtils.h [hunk 1]
```diff
@@ -8,7 +8,26 @@
 // See https://swift.org/LICENSE.txt for license information
 // See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 //
-// These utilities live in SIL/ so they be used by SIL verification.
+// These utilities model formal memory access locations as marked by
```
Comment (gottesmm) [inline include/swift/SIL/MemAccessUtils.h]: Can you make this match what LLVM does for file level comments? https://llvm.org/docs/CodingStandards.html#file-headers

**Review by gottesmm — APPROVED** [review]
LGTM modulo that one change.

File: include/swift/SIL/MemAccessUtils.h (hunk 1 above)
Comment (atrick) [reply include/swift/SIL/MemAccessUtils.h]: Yep

**Comment by atrick** [comment]
@swift-ci smoke test and merge

**Pull request merged**